app/controllers/v0/sign_in_controller.rb

```diff
@@ -181,25 +184,26 @@ def logout # rubocop:disable Metrics/MethodLength
         raise SignIn::Errors::MalformedParamsError.new message: 'Client id is not valid'
       end
 
-      unless load_user(skip_expiration_check: true)
-        raise SignIn::Errors::LogoutAuthorizationError.new message: 'Unable to Authorize User'
+      unless access_token_authenticate(skip_error_handling: true)
+        raise SignIn::Errors::LogoutAuthorizationError.new message: 'Unable to authorize access token'
       end
 
+      session = SignIn::OAuthSession.find_by(handle: @access_token&.session_handle)
+      credential_type = session.user_verification.credential_type
+
       SignIn::SessionRevoker.new(access_token: @access_token, anti_csrf_token:).perform
       delete_cookies if token_cookies
 
       sign_in_logger.info('logout', @access_token.to_s)
       StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_SUCCESS)
 
-      logout_redirect = SignIn::LogoutRedirectGenerator.new(user: @current_user,
+      logout_redirect = SignIn::LogoutRedirectGenerator.new(credential_type:,
                                                             client_config: client_config(client_id)).perform
-
       logout_redirect ? redirect_to(logout_redirect) : render(status: :ok)
     rescue SignIn::Errors::LogoutAuthorizationError, SignIn::Errors::SessionNotAuthorizedError => e
       sign_in_logger.info('logout error', { errors: e.message })
       StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_FAILURE)
-
-      logout_redirect = SignIn::LogoutRedirectGenerator.new(user: @current_user,
+      logout_redirect = SignIn::LogoutRedirectGenerator.new(credential_type: nil,
```

Comment:
Can you move the default `credential_type: nil` into the `LogoutRedirectGenerator` initializer? That way you can just call it without a `credential_type` param here